doc/dev_guide/internals/iproto/replication.rst

```diff
@@ -53,174 +49,94 @@ General
             -   Register an anonymous replica so it is not anonymous anymore
             
-Note that the master often sends :ref:`heartbeat <heartbeat>` messages to the replicas.
+The master also sends :ref:`heartbeat <heartbeat>` messages to the replicas.
 The heartbeat message's IPROTO_REQUEST_TYPE is ``0``.
 
-Synchronous
------------
+Below are details on individual replication requests.
+For synchronous replication requests, see :ref:`below <internals-iproto-replication-synchronous>`.
 
-..  container:: table
+..  _box_protocol-heartbeat:
 
-    ..  list-table::
-        :widths: 40 20 40
-        :header-rows: 1
+Heartbeats
+~~~~~~~~~~
 
-        *   -   Name
-            -   Code
-            -   Description     
+Once in :ref:`replication_timeout <cfg_replication-replication_timeout>` seconds,
+a master sends a :ref:`heartbeat <heartbeat>` message to a replica.
+The heartbeat message's IPROTO_REQUEST_TYPE is ``0``.
 
-        *   -   IPROTO_RAFT
-            -   0x1e
-            -   
-   
-        *   -   IPROTO_RAFT_PROMOTE
-            -   0x1f
-            -   Wait, then choose new replication leader. See :ref:`box.ctl.promote() <box_ctl-promote>`
+In the example below, the master sends a heartbeat message to a replica with id = 2,
+and the timestamp is a moment in 2020. The replica sends a response.
+Note that the master's heartbeat has no body:
 
-        *   -   IPROTO_RAFT_DEMOTE
-            -   0x20
-            -   
+..  raw:: html
+    :file: images/repl_heartbeat.svg
 
-        *   -   IPROTO_RAFT_CONFIRM
-            -   0x28
-            -
+IPROTO_TIMESTAMP is a float-64 MP_DOUBLE 8-byte timestamp.
 
-        *   -   IPROTO_RAFT_ROLLBACK
-            -   0x29
-            -
+The tutorial :ref:`Understanding the binary protocol <box_protocol-illustration>`
+shows actual byte codes of the above heartbeat examples.
 
+..  _box_protocol-join:
 
-..  code-block:: lua
+IPROTO_JOIN
+~~~~~~~~~~~
 
-    IPROTO_JOIN = 0x41 -- for replication
-    IPROTO_SUBSCRIBE = 0x42 -- for replication SUBSCRIBE
-    IPROTO_VOTE_DEPRECATED = 0x43 -- for old style vote, superseded by IPROTO_VOTE
-    IPROTO_VOTE = 0x44 -- for master election
-    IPROTO_FETCH_SNAPSHOT = 0x45 -- for starting anonymous replication
-    IPROTO_REGISTER = 0x46 -- for leaving anonymous replication.
+Code: 0x41.
 
+To join a replica set, an instance must send an initial IPROTO_JOIN request to any node in the replica set:
 
-Details on individual requests
-------------------------------
+..  raw:: html
+    :file: images/repl_join_request.svg
 
-..  _box_protocol-heartbeat:
+The node that receives the request sends its vclock in response:
 
-Heartbeats
-~~~~~~~~~~
+..  raw:: html
+    :file: images/repl_join_response.svg
 
-Frequently a master sends a :ref:`heartbeat <heartbeat>` message to a replica.
-For example, if there is a replica with id = 2,
-and a timestamp with a moment in 2020, a master might send this:
+After that, the node sends its last SNAP file,
```

Comment:
Not really. It looks like a SNAP file, but it's a snapshot of the current node state. (A snap file may be created an hour ago, and what is actually sent represents the fresh state)